### DISCUSSION TITLE
Fast Wipe

### DISCUSSION THREAD

**Discussion by oldtima** [FarGroup/FarManager#813] [Ideas]
Can you please add a fast wipe option only wiping the first sector or the first cluster in each file but with all the other goodies like resetting its size, dates and name (preferably to a single character) before deleting it? There is very rarely the need for a full wipe and most files are compressed these days thus becoming unidentifiable and irrecoverable if even the first few bytes are missing. I've been patching the source code and rebuilding it myself to reduce wiping in my Far to the first sector, but it could be a useful option for everyone.

**Comment by alabuzhev** [comment]
> most files are compressed these days thus becoming unidentifiable and irrecoverable if even the first few bytes are missing


[2.5 LZNT1 Algorithm Details](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-xca/5655f4a3-6ba4-489b-959f-e1f407c52f15):


> The LZNT1 buffer is structured as a series of chunks that can be _independently decompressed_.

[2.5.1.1 Buffer Format](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-xca/124d9696-a69c-409a-a055-2562fbe255f9):

> A compressed data buffer consists of one or more chunks. A chunk is either compressed, _uncompressed_, or it denotes the end of the buffer. If the chunk is uncompressed, it contains a chunk header followed by _uncompressed data_

**Reply by oldtima** [reply]
1. In 99% of the situations, nobody cares enough and nobody has access to the expensive rare software only available to the FSB or the NSA that takes months to write to identify random-looking clusters as potential chunks compressed by every single possible compression tool. What to talk of VHDX/PDF/WMA/MP3/MP4/AVI/MKV/.../.../...

2. After resetting file size to 0, that file's cluster list is removed from the MFT losing the file structure. Making this "attack" only applicable to non-fragmented files anyway.

3. SSD optimization (usually automatic and regularly performed) resets *all* the free clusters to 0 in a fraction of a second making all this file-by-file wiping a huge waste of time.

4. It doesn't take much to make it an option for those 99% of us who are not spies or criminals and merely need the file headers, names, sizes and dates disappear.

**Reply by bitraid** [reply]
As you already mentioned, the fact that SSD's do auto-TRIM, makes file wipe redundant.
You could just write a simple batch script (that you could call by macro or menu), to zero size/rename/delete.
If you still want to erase the headers, you could also add some powershell commands.
Example (save as `.cmd`): 
```cmd
@echo off
set F=%~f1
set R=%RANDOM%
set W=8192

:ask
if not exist "%F%" echo:File not found: "%F%"&goto:eof
set /p A=Fast wipe "%F%" (y/n)
if "%A%" == "n" goto:eof
if not "%A%" == "y" goto:ask

powershell -nop -c^
	$ErrorActionPreference = 'Stop';^
	$fStream = [System.IO.File]::Open('%F%', [System.IO.FileMode]::Open, [System.IO.FileAccess]::Write, [System.IO.FileShare]::None);^
	$bWriter = New-Object System.IO.BinaryWriter($fStream);^
	$zeros = New-Object byte[] %W%;^
	$bWriter.BaseStream.Position = 0;^
	$bWriter.Write($zeros);^
	$fStream.Close()^
	&& echo:|set/p=>"%F%" & ren "%F%" "%R%" && del /f /q "%R%"
```
